Shared/Extensions/JellyfinAPI/BaseItemDto+VideoPlayerViewModel.swift

```diff
@@ -39,10 +39,34 @@ extension BaseItemDto {
 
         let response = try await userSession.client.send(request)
 
-        guard let matchingMediaSource = response.value.mediaSources?
-            .first(where: { $0.eTag == mediaSource.eTag && $0.id == mediaSource.id })
-        else { throw JellyfinAPIError("Matching media source not in playback info") }
+        if liveTVChannel {
+            var matchingMediaSource: MediaSourceInfo? = nil
+            if let responseMediaSources = response.value.mediaSources {
+                for responseMediaSource in responseMediaSources {
+                    if let openToken = responseMediaSource.openToken, let mediaSourceId = mediaSource.id {
+                        if openToken.contains(mediaSourceId) {
+                            matchingMediaSource = responseMediaSource
+                        }
+                    }
+                }
+            }
+            guard let matchingMediaSource else {
+                throw JellyfinAPIError("Matching media source not in playback info")
+            }
 
-        return try matchingMediaSource.videoPlayerViewModel(with: self, playSessionID: response.value.playSessionID!)
+            return try matchingMediaSource.videoPlayerViewModel(
+                with: self,
+                playSessionID: response.value.playSessionID!,
+                liveTVChannel: liveTVChannel
+            )
+        } else {
+            guard let matchingMediaSource = response.value.mediaSources?
+                .first(where: { $0.eTag == mediaSource.eTag && $0.id == mediaSource.id })
+            else {
+                throw JellyfinAPIError("Matching media source not in playback info")
+            }
+
+            return try matchingMediaSource.videoPlayerViewModel(with: self, playSessionID: response.value.playSessionID!)
+        }
```

Comment:
1. `BaseItemDto+VideoPlayerViewModel`'s `videoPlayerVideoModel` func is not successfully matching the media source

I first found that Live TV playback was failing because no `videoPlayerViewModel` was getting created- that func was throwing an error. After investigating, I found that the `videoPlayerVideoModel` func was attempting to match a `mediaSource` from the response value that had a matching `id` and `eTag` as the input mediaSource. But the response for a liveTV channel did not have any such matching value. I'm not familiar enough with the Jellyfin APIs to really understand what is going on here, but after inspecting the response, I did find that a `mediaSource` within the response did have the input mediaSource id value buried in a combined string (that seems to hold multiple ids) in the `openToken` field. So for my initial workaround, I'm checking this `openToken` field to see if it contains a match for the input `mediaSource` id. If found, this mediaSource is used . 

I'm guessing this isn't the most elegant or ideal way to locate the matching mediaSource. I'm also thinking maybe liveTV should have its own function for creating this view model rather than passing in the liveTVChannel

Reply:
@sy6sy2 If you have the ability to run the app from source code attached to the debugger, I'd be interested to know if you put a breakpoint on line 80 of `BaseItemDto+VideoPlayerViewModel.swift`, and then tried to open a live TV channel, whether it successfully hits that breakpoint or not. My only initial idea on your issue is that maybe no matching media source is found. 

I already had some concerns about this change I made here, so it wouldn't be a huge surprise if it was related to your issue.

Reply:
That's great news! We've at least narrowed down the problem. 

In my setup, I am using IPTV + ErsatzTV channels, combined with Xteve, which is then loaded as an m3u playlist in Jellyfin. And with my setup, this media source matching logic happens to work.

I implemented this logic for media source matching by putting in breakpoints and looking at the data I was working with, and I just happened to notice that the ID of the mediaSource I was looking for happened to exist within this `openToken` field, but combined with other IDs, so something like `id1_id2_id3`. I added logic to check if it exists within that, and if found, consider it a match. 

But this was was essentially luck- I don't even know what the `openToken` field is or why it was the only place I could find the ID I am looking for here. 

I'll need to gain a better understanding of this, but I may need to find someone with deeper knowledge of Jellyfin APIs or the data I'm working with here to come up with a more robust solution.